package/gluon-hoodselector/luasrc/usr/sbin/hoodselector

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/lua
+
+local bit = require('bit')
+local unistd = require('posix.unistd')
+local fcntl = require('posix.fcntl')
+local hoodutil = require('hoodselector.util')
+
+-- PID file to ensure the hoodselector isn't running parallel
+local lockfile = '/var/lock/hoodselector.lock'
+local lockfd, err = fcntl.open(lockfile, bit.bor(fcntl.O_WRONLY, fcntl.O_CREAT), 384) -- mode 0600
+
+if not lockfd then
+  io.stderr:write(err, '\n')
+  os.exit(1)
+end
+
+local ok, _ = fcntl.fcntl(lockfd, fcntl.F_SETLK, {
+  l_start = 0,
+  l_len = 0,
+  l_type = fcntl.F_WRLCK,
+  l_whence = unistd.SEEK_SET,
+})
+
+if not ok then
+  io.stderr:write(string.format(
+    "Unable to lock file %s. Make sure there is no other instance of the hoodselector running.\n",
+    lockfile
+  ))
+  os.exit(1)
+end
+
+-- VPN MODE
+-- If we have a VPN connection we will try to get the node's location and
+-- select the hood coresponding to this location.
+-- If no hood for the location has been defined, we will select
+-- the default hood.
+-- If we can't determine the node's location, we will continue to the next mode.
+if hoodutil.directVPN() then
+  io.stdout:write('VPN connection found.\n')
+  local geo = hoodutil.getGeolocation()
+  if geo.lat ~= nil and geo.lon ~= nil then
+    io.stdout:write('Position found.\n')
+    local jhood = hoodutil.get_domains()
+    local geoHood = hoodutil.getHoodByGeo(jhood, geo)
+    if geoHood ~= nil then
+      if hoodutil.set_hoodconfig(geoHood) then
+        hoodutil.restart_services()
+        io.stdout:write('Hood set by VPN mode.\n')
+      end
+      os.exit(0)
+    end
+    io.stdout:write('No hood has been defined for the current position.\n')
+    if hoodutil.set_hoodconfig(hoodutil.getDefaultHood(jhood)) then
+      hoodutil.restart_services()
```

Comment:
There should be a way to run this so it only updates the configured domain, but does not restart anything. This way we could run it once in config mode after the location has been set (which makes a lot more sense than running it from cron in setups without dynamic updates of the location).

Reply:
Maybe I can check if the UCI parameters for setup_mode.enabled ?
But I think you miss understood something, the applying of new domain configurations could not be done by this mode while the node is in config mode.

Reply:
@NeoRaider are you fine with that solution?